src/features/deposit/components/Form/DepositFormNetwork/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { useForm } from "react-hook-form"
+
+import { Button, Spinner, Text } from "@radix-ui/themes"
+import { Form } from "../../../../../components/Form"
+import { Select } from "../../../../../components/Select/Select"
+import { BlockchainEnum } from "../../../../../types/deposit"
+import styles from "./styles.module.css"
+
+export type DepositFormNetworkValues = {
+  blockchain: BlockchainEnum
+}
+
+export interface DepositFormNetworkProps {
+  options: {
+    [key in BlockchainEnum]: {
+      label: BlockchainEnum
+      icon: React.ReactNode | null
+    }
+  }
+  onSubmit: (values: DepositFormNetworkValues) => void
+}
+
+export const DepositFormNetwork = ({
+  options,
+  onSubmit,
+}: DepositFormNetworkProps) => {
+  const {
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm<DepositFormNetworkValues>({ reValidateMode: "onSubmit" })
```

Comment:
I don't think we need form here, because we don't do anything with form (validate, submit, store data in fields) we can simplify it and just store blockchain in useState.

Reply:
Well we do submit and store select in a filed, so form my point of view it is still a form, even without validation. The key point is that it uses `reValidateMode` on submit, which ensures that if a user navigates back from a sibling form, we don’t need to worry about resetting the useState constant.